docs/visualizing-your-sql-queries/plotting-with-seaborn.md

```diff
@@ -1,14 +1,559 @@
+```
+
+```{code-cell} ipython3
+%sqlcmd explore --table s1.district
+```
+
+```{code-cell} ipython3
+%sqlcmd explore --table s1.loan
+```
+
+## Matplotlib inheritance
+
+Seaborn is built on top of Matplotlib. Therefore, depending on the seaborn plotting command, it will return either a Matplotlib axes or figure object. If the plotting function is axes-level, a single `matplotlib.pyplot.Axes` object is returned. This object accepts an `ax=` argument, which integrates with Matplotlib's <b>object-oriented interface</b> and allows composing plots with other plots. On the other hand, if the plotting function is figure-level, a `FacetGrid` object is returned. This object, unlike the axes-level object, is more standalone, but "smart" about subplot organization. To learn about these objects in greater detail, visit seaborn's website [here](https://seaborn.pydata.org/tutorial/function_overview.html#figure-level-vs-axes-level-functions).
+
+A few examples denoting this distinction are shown below.
+
+### Axes-level
+
+Suppose we want to identify whether gentrification increases the average salary of two regions, using the data we downloaded above. We first save our [CTE (Common Table Expression)](https://ploomber-sql.readthedocs.io/en/latest/advanced-querying-techniques/ctes) named `levels_example` that takes in the columns, `average_salary`, `ratio_of_urban_inhabitants`, and `region`, and filters for two regions, 'central Bohemia' and 'east Bohemia', from the `s1.district` table.
+
+```{code-cell} ipython3
+%%sql --save levels_example
+SELECT average_salary, ratio_of_urban_inhabitants, region
+FROM s1.district
+WHERE region IN ('central Bohemia', 'east Bohemia');
+```
+
+The result of the CTE is saved as a pandas `DataFrame()`:
+
+```{code-cell} ipython3
+result = %sql SELECT * FROM levels_example
+df = result.DataFrame()
+```
+
+You can determine what is returned using Python's `type()` function:
+
+```{code-cell} ipython3
+plt.rcParams["figure.dpi"] = 300  # high resolution
+
+scatter_plt = sns.scatterplot(
+    data=df, x="ratio_of_urban_inhabitants", y="average_salary", hue="region"
+)
+print(type(scatter_plt))
+```
+
+Notice that the `sns.scatterplot()` function returns a Matplotlib `Axes` object, a single plot that is inclusive of the data from both regions. Other seaborn functions, including `regplot`, `boxplot`, `kdeplot`, and many others, also return Matplotlib `Axes` objects. Therefore, we can use various Matplotlib axes [commands](https://seaborn.pydata.org/generated/seaborn.axes_style.html) to modify the Seaborn figure.
+
+### Figure-level
+
+On the other hand, we can show that `sns.relplot()` returns a `FacetGrid` object, creating separate plots for the regions:
+
+```{code-cell} ipython3
+facet_plt = sns.relplot(
+    data=df, x="ratio_of_urban_inhabitants", y="average_salary", col="region"
+)
+print(type(facet_plt))
+```
+
+Other Figure-level seaborn functions include `catplot`, `displot`, `pairplot`, and `jointplot`.
```

Comment:
capitalization seems off for Figure-level

Reply:
changing the "F" to lower case